unit-09-try-except/exercise.ipynb

```diff
@@ -0,0 +1,323 @@
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "### Example: Error handling with `else` block"
+      ],
+      "metadata": {
+        "id": "yDgA816z9I0H"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "try:\n",
+        "    x = 10 / 2\n",
+        "except ZeroDivisionError:\n",
+        "    print(\"You can't divide by zero!\")\n",
+        "else:\n",
+        "    print(\"Division was successful.\")"
+      ],
+      "metadata": {
+        "id": "CvXKK5Kp7EL-"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "try:\n",
+        "    x = 10 / 0\n",
+        "except ZeroDivisionError:\n",
+        "    print(\"You can't divide by zero!\")\n",
+        "else:\n",
+        "    print(\"Division was successful.\")"
+      ],
+      "metadata": {
+        "id": "OA4HtiNd37nc"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "### Example: Error handling with `finally` block"
+      ],
+      "metadata": {
+        "id": "xykpCEoC9Ril"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "try:\n",
+        "    x = 10 / 0\n",
+        "except ZeroDivisionError:\n",
+        "    print(\"You can't divide by zero!\")\n",
+        "finally:\n",
+        "    print(\"Code in the finally block will always run\")\n"
+      ],
+      "metadata": {
+        "id": "ekSyhbnG6kMF"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "try:\n",
+        "    x = 10 / 2\n",
+        "except ZeroDivisionError:\n",
+        "    print(\"You can't divide by zero!\")\n",
+        "finally:\n",
+        "    print(\"Code in the finally block will always run\")"
+      ],
+      "metadata": {
+        "id": "R6LLgY9r6l0x"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "### Exercise 2\n",
+        "`else` or `finally`? Which block should be used?"
+      ],
+      "metadata": {
+        "id": "ffW6zAVJ9lR5"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "string1 = \"abc\"\n",
+        "try:\n",
+        "    bad_code = int(string1) # unsuccessful attempt to convert string to an integer\n",
+        "except ValueError:\n",
+        "    print(f\"{string1} cannot be converted into an integer\")\n",
+        "# ?????:\n",
+        "#     print(\"This statement should always run\")"
+      ],
+      "metadata": {
+        "id": "SW0M0CYU9dRQ"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "string2 = \"1\"\n",
+        "try:\n",
+        "    bad_code = int(string2) # unsuccessful attempt to convert string to an integer\n",
```

Comment:
I changed "unsuccessful" to "successful" since we can successfully convert string "1" to an integer.

```suggestion
        "string2 = \"1\"\n",
        "try:\n",
        "    bad_code = int(string2) # successful attempt to convert string to an integer\n",
```